sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -544,23 +514,29 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
    * @throws `RestError` with code that is a value from the standard set of HTTP status codes as documented at
    * https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.8
    */
-  async updateQueue(queueName: string, queueOptions: QueueOptions): Promise<UpdateQueueResponse> {
+  async updateQueue(queue: QueueDescription): Promise<UpdateQueueResponse> {
     log.httpAtomXml(
-      `Performing management operation - updateQueue() for "${queueName}" with options: ${queueOptions}`
+      `Performing management operation - updateQueue() for "${queue.queueName}" with options: ${queue}`
     );
 
-    if (!isJSONLikeObject(queueOptions) || queueOptions === null) {
+    if (!isJSONLikeObject(queue) || queue === null) {
       throw new TypeError(
-        `Parameter "queueOptions" must be an object of type "QueueOptions" and cannot be undefined or null.`
+        `Parameter "queue" must be an object of type "QueueDescription" and cannot be undefined or null.`
       );
     }
 
-    const finalQueueOptions: QueueOptions = {};
-    const getQueueResult = await this.getQueueDetails(queueName);
-    Object.assign(finalQueueOptions, getQueueResult, queueOptions);
+    if (areOptionsUndefined(queue, "queueName")) {
+      throw new TypeError(
+        `Parameter "queue" must be an object of type "QueueDescription" and at least one of the parameters other than queueName must be defined.`
```

Comment:
Why do we care about this? What happens if we let such requests reach the service? And why is this check made only for queues, but not topics or subscriptions?

Reply:
> I had added this error which seemed to make sense since there is no need to make the update request when there are no options provided

Yes, but we don't need to be validating user input as long as we can make a request to the service. To make the request, I believe all we need is the queue name, so we could add a check for that.

Reply:
Removed this additional check of "at least one option other than the entityName" and instead added "queueName" check(similarly for others)